idea/Hublsoft.Task.Web/Areas/Identity/Pages/Account/Login.cshtml.cs

```diff
@@ -82,6 +85,7 @@ public async Task<IActionResult> OnPostAsync(string returnUrl = null)
                 var result = await _signInManager.PasswordSignInAsync(Input.Email, Input.Password, Input.RememberMe, lockoutOnFailure: false);
                 if (result.Succeeded)
                 {
+                    _service.RecordLogin(Input.Email, DateTime.Now);
```

Comment:
Agree, Identity Servers events mechanism is a good clean way to handle this kind of event without upsetting the flow. As i mentioned earlier, if looking to implement a real world solution i would have no hesitation in using Identity Server.

Reply:
> Also, something to think about is what happens when a user registers. I think currently when a client registers, they are not automatically logged in. It would be good to ensure that all sign ins result in a login event being recorded. Maybe encapsulating the `SignInManager` might be a good way of ensuring that?

The break in the "register - login" process is intended as a barrier to prevent unauthorised access when using email address as the user identifier. This allows the authenticating/validation of the users email address by sending a confirmation email. 
However i do agree that this implementation needs to be expanded, i would like to include all login events successful or failures (unrecognised account/incorrect password etc)

Reply:
Yeah certainly in a real world scenario these would be useful